src/chain.ts

```diff
@@ -0,0 +1,514 @@
+/*
+ * ISC License (ISC)
+ * Copyright (c) 2022 aeternity developers
+ *
+ *  Permission to use, copy, modify, and/or distribute this software for any
+ *  purpose with or without fee is hereby granted, provided that the above
+ *  copyright notice and this permission notice appear in all copies.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+ *  REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+ *  AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+ *  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+ *  LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+ *  OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+ *  PERFORMANCE OF THIS SOFTWARE.
+ */
+
+/**
+ * Chain module
+ * @module @aeternity/aepp-sdk/es/chain
+ * @export Chain
+ * @example import { Chain } from '@aeternity/aepp-sdk'
+ */
+
+import { AE_AMOUNT_FORMATS, formatAmount } from './utils/amount-formatter'
+// @ts-expect-error TODO remove
+import verifyTransaction from './tx/validator'
+import { pause } from './utils/other'
+// @ts-expect-error TODO remove
+import { isNameValid, produceNameId, decode } from './tx/builder/helpers'
+// @ts-expect-error TODO remove
+import { DRY_RUN_ACCOUNT } from './tx/builder/schema'
+// @ts-expect-error todo remove
+import Node from './node'
+// @ts-expect-error todo remove
+import Ae from './'
+import {
+  AensPointerContextError,
+  DryRunError,
+  InvalidAensNameError,
+  InvalidTxError,
+  RequestTimedOutError,
+  TxTimedOutError,
+  TxNotInChainError
+} from './utils/errors'
+
+interface Transaction {
+  hash: string
+  rawTx: string
+  confirmationHeight?: number
+}
+
+interface Account {
+  balance: string
+  id: string
+  kind: string
+  nonce: number
+  payable: boolean
+}
+
+interface Tx {
+  blockHash: string
+  blockHeight: number
+  hash: string
+  signatures: string[]
+  tx: any
+}
+
+interface Generation {
+  keyBlock: {
+    beneficiary: string
+    hash: string
+    height: number
+    info: string
+    miner: string
+    nonce: string
+    pow: number[]
+    prevHash: string
+    prevKeyHash: string
+    stateHash: string
+    target: number
+    time: number
+    version: number
+  }
+  microBlocks: any[]
+}
+
+interface ReqTransaction {
+  tx: string
+  accountAddress: string
+  top: string | number
+  txEvents: boolean
+  options?: any
+  resolve: Function
+  reject: (error: any) => void
+}
+
+interface Contract {
+  abiVersion: number
+  active: boolean
+  deposit: number
+  id: string
+  ownerId: string
+  referrerIds: any[]
+  vmVersion: number
+}
+
+export function _getPollInterval (
```

Comment:
```suggestion
function _getPollInterval (
```

Seems like this method is not imported anywhere else and it also follows the private naming convention